### ISSUE TITLE
Fluent 1.2.0: Autentication failed in AzureEnvironment.AzureGermanCloud: wrong audience or resource

### ISSUE THREAD

**Issue by dvdgutzmann** [Azure/azure-sdk-for-net#3656]
Hello *

I'm getting authentication failures when trying to access the AzureGermanCloud using the 1.2.0 fluent SDK. 

Packages installed are Microsoft.Azure.Management.Fluent 1.2.0 and dependencies. Target framework is .NET 4.7.  I'm using Visual Studio 2017 Update 3.

> The access token has been obtained from wrong audience or resource 'https://portal.microsoftazure.de/'. It should exactly match (including forward slash) with one of the allowed audiences 'https://management.core.cloudapi.de/','https://management.microsoftazure.de/'.

```csharp
var credentials = SdkContext.AzureCredentialsFactory.FromServicePrincipal(
        clientId,
        clientSecret,
        tenantId,
        AzureEnvironment.AzureGermanCloud
);

var azure = Azure.Configure().Authenticate(credentials).WithSubscription(subscriptionId);

var vm = azure.VirtualMachines.GetByResourceGroup("Common", "jumphost");
```

The ServicePrincipal was created using PowerShell

```powershell
$app = New-AzureRmADApplication -DisplayName "*" -HomePage "*" -IdentifierUris "*" -Password "*"
New-AzureRmADServicePrincipal -ApplicationId $app.ApplicationId
New-AzureRmRoleAssignment -RoleDefinitionName Contributor -ServicePrincipalName $app.ApplicationId.Guid
```

Is there anything I should do differently in the AzureGermanCloud? 

I'm happy to provide any information necessary to resolve this issue as quickly as possible. I'm also going to open a support case regarding this  issue, as we need a solution as soon as possible.

Thank you!

Kind regards,

David Gutzmann

**Comment by martijnkamphuis** [comment]
Can confirm the issue. A temporary work around for me is to set the AzureEnvironment.AzureGermanCloud.ManagementEnpoint (the missing 'd' typo in Endpoint is on purpose, since it's in the SDK also) to "https://management.microsoftazure.de/".

**Comment by dvdgutzmann** [comment]
Hello @martijnkamphuis, 

thank you for providing the workaround! It's working for me when setting the ManagementEnpoint (sic!) accordingly.

```csharp
var azureCloud = AzureEnvironment.AzureGermanCloud;
azureCloud.ManagementEnpoint = "https://management.microsoftazure.de/";

var credentials = SdkContext.AzureCredentialsFactory.FromServicePrincipal(clientId, clientSecret, tenantId, azureCloud);
```

**Comment by dvdgutzmann** [comment]
Case id with MS support is 117090516284101

**Comment by hovsepm** [comment]
Fixed. Will be shipped in release 1.3
